src/commands/run.ts

```diff
@@ -405,18 +406,54 @@ const fluenceRun = async (args: RunArgs) => {
   }
 
   await initFluenceClient(args, args.maybeFluenceConfig);
-  const { Fluence, callAquaFunction } = await import("@fluencelabs/js-client");
+
+  const { Fluence, callAquaFunction, js2aqua, aqua2js } = await import(
+    "@fluencelabs/js-client"
+  );
+
+  const schema = functionCall.funcDef;
+
+  // TODO: remove this after DXJ-535 is done
+  const validatedRunData = Object.fromEntries(
+    Object.entries(args.runData ?? {}).map(([argName, argValue]) => {
+      const fields =
+        schema.arrow.domain.tag === "nil" ? {} : schema.arrow.domain.fields;
+
+      const argSchema = fields[argName];
+
+      assert(
+        argSchema !== undefined,
+        "Should be impossible because schema always contains same keys as runData",
```

Comment:
It's totally possible. User can put anything in runData even accidentally use incorrect name. We should validate this and give a good error message if user does something wrong